### ISSUE TITLE
Texture splatting!!!

### ISSUE THREAD

**Issue by Thundros** [ol-bones/Three.js-Game-Engine#33]
Hi @ol-bones ! First off I'd like to say : GOOD JOB! I LOVE this! PLEASE keep up the GREAT work! <3 Second, I'd like to ask if you can add Texture Splatting to Terrain & ability to export Terrain WITH Texture Splatting Quality applied to the model?

Once again, PLEASE keep up the GREAT work & I look forward to beta testing this! <3

**Comment by Thundros** [comment]
@ol-bones : Hello?

**Comment by ol-bones** [comment]
I'm working on this atm, glad you like it haha

**Comment by Thundros** [comment]
Hey @ol-bones : How goes the Texture splatting / saving? <3

**Comment by ol-bones** [comment]
Like this? grass/mud

https://i.imgur.com/tYCUc1E.png

**Comment by Thundros** [comment]
@ol-bones : <3 _ <3

**Comment by Thundros** [comment]
Something like this : 

https://www.blog.radiator.debacle.us/2013/09/hacking-blend-transition-masks-into.html

This is an example splatmap I have drawn in Paint.net : 

https://imgur.com/Jjn3oVd

**Comment by ol-bones** [comment]
https://i.imgur.com/wXz0rB3.mp4

**Comment by Thundros** [comment]
@ol-bones : O my! You are TOO good! :O Can you make it so that you can use any size texture other than 0? HOLY! Keep up the AWESOME work! <3

**Comment by ol-bones** [comment]
> Can you make it so that you can use any size texture other than 0?

wdym?

[painted terrain](https://github.com/ol-bones/Three.js-Game-Engine/blob/master/resource-api/content/entities/terrain/PaintedTerrain/Painted%20Terrain.json)
```

                "Size": 1000,
                "Divisions": 25,
                "material": {
                    "type": "TerrainMapMaterial",
                    "texture": "/grass2.jpg",
                    "texture2": "/dirt.jpg",
                    "color": "0xFFFFFF",
                    "repeat": [
                        9,
                        9
                    ],
```

texture1 and 2 can be anything any size

**Comment by arpu** [comment]
what is the maximal different Textures 4 ?

**Comment by ol-bones** [comment]
2 with simple mix() shader

**Comment by Thundros** [comment]
@ol-bones : O_O THAT'S AMAZING! <3 Is there a way to save the terrain with the new quality textures already applied to terrain?

**Comment by ol-bones** [comment]
@Thundros Yeah make a new entity like PaintedTerrain but with the new textures and call it something else, I haven't added an option in the menus to change the textures on terrains

I'm also not going to work on this anymore probably. Mostly because I've found cannonjs isn't really efficient enough for a lot of physics objects. It's probably ok to use this to make a game without physics like Divinity Original Sin but FPS needs to be smooth.

Feel free to use the code or reference it in your own stuff

**Comment by Thundros** [comment]
@ol-bones : Do NOT give up! Use ammo.js

**Comment by ol-bones** [comment]
Yep! Ammo.js is probably better, I don't have time for this anymore though haha.

**Comment by Thundros** [comment]
@ol-bones : :(

**Comment by ol-bones** [comment]
@Thundros life :(

**Comment by Thundros** [comment]
@ol-bones : Is there somebody you can give the source to who CAN actively work on it? I don't wan to see this project fail.

**Comment by Thundros** [comment]
@ol-bones : Why not start a kickstarter?

**Comment by ol-bones** [comment]
@Thundros  Nope, feel free to add to it yourself though, or post on www.reddit.com/r/threejs or www.reddit.com/r/gamedev - maybe someone there would contribute

Also the average kickstarter funds raised is $5k to $23k - so not a very viable option

Glad you like it so much though haha

**Comment by Thundros** [comment]
@ol-bones : What about GoFundMe?

**Comment by ol-bones** [comment]
@Thundros Make it for me, if it gets £500K I'll finish it and make a sick game

**Comment by Thundros** [comment]
@ol-bones : Make what?

**Comment by ol-bones** [comment]
a gofundme page

**Comment by Thundros** [comment]
You don't know how to create one?

**Comment by ol-bones** [comment]
well yeah but it's your idea and you're passionate about it so it's probably better for you to do it
